### DISCUSSION TITLE
Cannot set keyboard in crocods.ini

### DISCUSSION THREAD

**Discussion by mtrevor** [eduardofilo/RG350_adam_image#294] [RetroArch topics]
### What happened?

I'm trying to change the `key/emulation` setting in `/media/data/local/home/.crocods/crocods.ini` from 3 to 1 in an attempt to switch the keyboard input from azerty to qwerty. However, when I run CrocoDS again, the input is still azerty and the ini file is set back to 3.

I couldn't see an option in CrocoDS to set the keyboard type and couldn't find much in the way of CrocoDS docs. Are configs in the home folder modifiable?

Cheers.

### Device

RG280V

### Version

v2.1

**Comment by a-dekker** [comment]
Can't really help you, but I can confirm that the key -> emulation value is always reset to '3'

Unfortunately the code at https://github.com/redbug26/crocods-core is outdated so I can't tell why and what this option is supposed to be doing.

The standalone version http://crocods.org/crocods_20200115.opk has a proper qwerty keybinding, so it seems not related to this key setting. 

Certain config values can be changed, sound enabled/disabled works for example.

You could try to open an issue at https://github.com/redbug26/crocods-core/issues or somewhere at libretro perhaps.

**Comment by eduardofilo** [comment]
Good answer @a-dekker.

I have tested and in my case the qwerty keyboard appears whether or not that setting is present (in the newly installed image that configuration file contains nothing but comments).

![screenshot001](https://user-images.githubusercontent.com/2923728/189489612-06388282-f8ae-4b36-a067-15756e600cb2.png)

In any case I turn the issue into a discussion since it is an emulator problem.

EDIT: OK, I have seen that although the template displayed on the screen is always qwerty, the keyboard does indeed behave like azerty when the keys are pressed. I had misinterpreted the problem.

**Comment by a-dekker** [comment]
Just noticed the issue is reported about a year ago at libretro : https://github.com/libretro/libretro-crocods/issues/18

**Comment by a-dekker** [comment]
And just found out the reason: https://github.com/libretro/libretro-crocods/blob/master/libretro.c#L164
It is hardcoded set to French (1). When set to 0 (default), it will switch to qwerty. Also reported at https://github.com/libretro/libretro-crocods/issues/18
